cmds/dhclient/dhclient.go

```diff
@@ -75,17 +76,22 @@ func dhclient4(iface netlink.Link, numRenewals int, timeout time.Duration) error
 		debug("Start getting or renewing DHCPv4 lease")
 
 		var success bool
-		if packet == nil {
-			success, packet, err = client.Request()
-		} else {
-			success, packet, err = client.Renew(packet)
-		}
-		if err != nil {
-			networkError, ok := err.(*net.OpError)
-			if ok && networkError.Timeout() {
-				return fmt.Errorf("%s: could not find DHCP server", mac)
+		for i := 0; i < retry || retry < 0; i++ {
+			if packet == nil {
+				success, packet, err = client.Request()
+			} else {
+				success, packet, err = client.Renew(packet)
```

Comment:
This doesn't work with packet. Next round, packet will be nil if this has an error, so then client.Request is called after client.Renew.